### PR TITLE
Fix Compatibility Tests using govv1beta1 GenesisStates

### PR DESCRIPTION
**Pull request by chatton**
## Description

We ran into an issue where the tests were incorrectly determining if the chain should be using govv1beta1 or govv1 genesis state. This caused startup failures with unrecognised fields (params).

This PR splits the two options into two separate functions, and now correctly handles the release branch e.g. `release-v4.0.x`

We can definitely come up with a better long term solution, but I went with this way for now as we need it to be able to run the compatibility tests.

v7.0.0 compatibility tests: ~https://github.com/cosmos/ibc-go/actions/runs/4073293413~ (gov test failed)
After gov fix: https://github.com/cosmos/ibc-go/actions/runs/4073838154

### Commit Message / Changelog Entry

N/A

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/semverutil/semver.go [hunk 1]
```diff
@@ -16,6 +18,17 @@ type FeatureReleases struct {
...
+
+	// in our compatibility tests, our images are in the format of "release-v1.0.x". We want to actually look at
+	// the "1.0.x" part but we also need this to be a valid version. We can change it to "1.0.0"
+	// TODO: change the way we provide the ibc-go version. This should be done in a more flexible way such
+	// as docker labels/metadata instead of the tag, as this will only work for our versioning scheme.
+	const releasePrefix = "release-"
+	if strings.HasPrefix(versionStr, releasePrefix) {
+		versionStr = versionStr[len(releasePrefix):]
+		versionStr = strings.ReplaceAll(versionStr, "x", "0")
+	}
```
Comment (damiannolan) [inline e2e/semverutil/semver.go:30]: Would the labels approach be something like adding the `LABEL` cmd to the dockerfile.
Something like:

```
LABEL ibc-go=v7.0.0
LABEL cosmos-sdk=v0.47.0
```

with some smarter way of retrieving the versions. I haven't looked at docker labels a whole lot, there may be a way to add labels after the image is built too

**Review by damiannolan — APPROVED** [review]
Nice, lgtm! Hopefully we will have less disruptions with genesis versions etc moving forward :)

File: e2e/semverutil/semver.go (hunk 1 above)
Comment (chatton) [reply e2e/semverutil/semver.go:30]: yeah that syntax should work, or using `docker build --label`. It's not possible to add labels to an existing container, but they could be re-built as necessary to facilitate this.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -58,4 +58,4 @@ jobs:
         push: true
         tags: ${{ steps.meta.outputs.tags }}
         build-args: |
-          IBC_GO_VERSION="${{ github.ref_name }}"
+          IBC_GO_VERSION=${{ github.ref_name }}
```
Comment (chatton) [inline .github/workflows/release.yml:61]: same fix that was already applied in the 4.3.x release branch but was still in main.

File: e2e/testsuite/testsuite.go [hunk 3]
```diff
@@ -506,6 +512,11 @@ func (s *E2ETestSuite) ExecuteGovProposalV1(ctx context.Context, msg sdk.Msg, ch
 	msgSubmitProposal, err := govtypesv1.NewMsgSubmitProposal(msgs, sdk.NewCoins(sdk.NewCoin(chain.Config().Denom, govtypesv1.DefaultMinDepositTokens)), sender.String(), "", fmt.Sprintf("e2e gov proposal: %d", proposalID), fmt.Sprintf("executing gov proposal %d", proposalID))
 	s.Require().NoError(err)
 
+	if !govv1ProposalTitleAndSummary.IsSupported(chain.Nodes()[0].Image.Version) {
+		msgSubmitProposal.Title = ""
+		msgSubmitProposal.Summary = ""
+	}
```
Comment (chatton) [inline e2e/testsuite/testsuite.go:518]: without zero'ing out these fields there is an `errorUnknownField` error when broadcasting this message to pre-sdk 47 chains.

**Pull request merged**